tests/test_activate.py

```diff
@@ -3099,3 +3099,110 @@ def test_activate_and_deactivate_for_uninitialized_env(conda_cli):
     assert conda_error.value.message.startswith(
         "Run 'conda init' before 'conda deactivate'"
     )
+
+
+# The MSYS2_PATH tests are slightly unusual in two regards: firstly
+# they stat(2) for potential directories which indicate which (of
+# several) possible MSYS2 environments have been installed; secondly,
+# conda will pass a Windows pathname prefix but conda-build will pass
+# a Unix pathname prefix (in particular, an MSYS2 pathname).
+
+
+def assert_MSYS2_PATH(create_dirs=None, exp_paths=None, no_paths=None):
+    with tempdir() as td:
+        mkdir_p(join(td, "conda-meta"))
+        activate_d_dir = mkdir_p(join(td, "etc", "conda", "activate.d"))
+        activate_d_1 = join(activate_d_dir, "see-me.sh")
+        activate_d_2 = join(activate_d_dir, "dont-see-me.bat")
+        touch(join(activate_d_1))
+        touch(join(activate_d_2))
+
+        activate_env_vars = join(td, PREFIX_STATE_FILE)
+        with open(activate_env_vars, "w") as f:
+            f.write(ENV_VARS_FILE)
+
+        write_pkg_env_vars(td)
```

Comment:
Should this be [`write_pkgs(td)`](https://github.com/conda/conda/blob/main/tests/test_activate.py#L186) or something else?

Reply:
See the previous comment about the code changing "a bit" in the last few weeks.  I'll need to revisit.